llm/src/types.ts

```diff
@@ -0,0 +1,76 @@
+import { LlmPrompt } from "./prompts/prompting.ts";
+
+export type LLMResponse = {
+  content: string;
+  // The trace span ID
+  id: string;
+};
+
+export type ModelName = string;
+export type LLMPrompt = LlmPrompt;
+export interface LLMTypedContent {
+  type: "text" | "image";
+  data: string;
+}
+export type LLMContent = string | LLMTypedContent[];
+export type LLMMessage = {
+  role: "user" | "assistant";
+  content: LLMContent;
+};
+export type LLMRequestMetadata = Record<string, string | undefined | object>;
```

Comment:
Could this always be `Record<string, string>`, do we need `undefined` and `object`? TBD

Reply:
The `LlmPrompt` object requires `object` metadata values